services/ui_backend_service/data/db/tables/run.py

```diff
@@ -135,57 +111,42 @@ def select_columns(self):
             WHEN end_attempt_ok IS NOT NULL
             THEN end_attempt_ok.ts_epoch
             WHEN {table_name}.last_heartbeat_ts IS NOT NULL
-                AND latest_failed_task IS NOT NULL
-                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)>{heartbeat_threshold}
-            THEN {table_name}.last_heartbeat_ts*1000
-            WHEN {table_name}.last_heartbeat_ts IS NOT NULL
-                AND latest_failed_task IS NULL
-                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)>{heartbeat_cutoff}
-            THEN {table_name}.last_heartbeat_ts*1000
-            ELSE NULL
+                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)<={heartbeat_threshold}
+            THEN NULL
+            ELSE {table_name}.last_heartbeat_ts*1000
         END) AS finished_at
         """.format(
             table_name=table_name,
             heartbeat_threshold=HEARTBEAT_THRESHOLD,
-            heartbeat_cutoff=RUN_INACTIVE_CUTOFF_TIME
+            heartbeat_cutoff=RUN_INACTIVE_CUTOFF_TIME,
         ),
         """
         (CASE
-            WHEN end_attempt_ok.value IS TRUE
-            THEN 'completed'
-            WHEN {table_name}.last_heartbeat_ts IS NOT NULL
-                AND latest_failed_task IS NOT NULL
-                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)>{heartbeat_threshold}
-            THEN 'failed'
-            WHEN end_attempt_ok.value IS FALSE
-                AND end_attempt.ts_epoch > end_attempt_ok.ts_epoch
+            WHEN end_attempt IS NOT NULL
+                AND end_attempt_ok.ts_epoch < end_attempt.ts_epoch
```

Comment:
this is the same type of logic as above just for the state instead of the finished_at time.